libicalvcard: escape unescaped backslash at end of value

rsto opened the pull request:
Fixes a bug where an unescaped backslash at the end of a value caused the vCard parser attempt to free the multi-valued TEXT buffer, even if that had not been allocated.

Rather than rejecting a TEXT property with a trailing backslash as invalid, the parser now pads that backslash with a second backslash, preserving the backslash character effectively.

brong commented on src/libicalvcard/vcardparser.c:
```diff
@@ -688,20 +688,26 @@ static int _parse_prop_value(struct vcardparser_state *state)
+                        /* value ends with unescaped backslash */
+                        PUTC('\\');
+                        PUTC('\\');
+                        INC(2);
```
why INC(2) here?  That eats two characters of input.

brong replied:
I have had this explained :)  Because we `goto out` we want to eat the \n as well because that's the line terminus.

brong reviewed (approved):
This looks heaps better, thanks.  It means we do sensible things for un-sensible input!

rsto committed:
libicalvcard: escape unescaped backslash at end of value

Fixes a bug where an unescaped backslash at the end of a value
caused the vCard parser attempt to free the multi-valued TEXT
buffer, even if that had not been allocated.

Rather than rejecting a TEXT property with a trailing backslash
as invalid, the parser now pads that backslash with a second
backslash, preserving the backslash character effectively.
